subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -850,6 +968,34 @@ void nct_dc_endpoint_set(const struct nrf_cloud_data *tx_endp,
 	if (m_endp != NULL) {
 		nct.dc_m_endp.utf8 = (u8_t *)m_endp->ptr;
 		nct.dc_m_endp.size = m_endp->len;
+
+#if defined(CONFIG_AWS_FOTA)
+		int ret;
+
+		nct.job_status_endp.size =
+			nct.dc_m_endp.size + NCT_TOPIC_PREFIX_M_D_LEN +
+			NRF_CLOUD_CLIENT_ID_LEN + NCT_JOB_STATUS_TOPIC_LEN + 1;
+		nct.job_status_endp.utf8 =
+			nrf_cloud_malloc(nct.job_status_endp.size);
+		if (nct.job_status_endp.utf8 == NULL) {
+			LOG_ERR("Failed to allocate mem for job status topic");
+			return;
+		}
+		ret = snprintf(nct.job_status_endp.utf8,
+			       nct.job_status_endp.size, "%s%s%s%s",
+			       nct.dc_m_endp.utf8, NCT_M_D_TOPIC_PREFIX,
+			       client_id_buf, NCT_JOB_STATUS_TOPIC);
+
```

Comment:
This empty line should be moved before `snprintf` call (no need to separate return value check).